### PR TITLE
484 proxy login

### PR DESCRIPTION
**Pull request by Sanilblank**

### CODE REVIEW

File: app/Console/Commands/CreateSuperAdmin.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?php
+
```
Comment (man-git007) [inline app/Console/Commands/CreateSuperAdmin.php:2]: declare strict type missing

File: app/Http/Controllers/Admin/Setting/SettingController.php [hunk 2]
```diff
@@ -263,6 +263,7 @@ public function getSettingStatus(): JsonResponse
                 'data' => $status,
             ]);
         } catch (\Exception $e) {
+            dd($e);
```
Comment (man-git007) [inline app/Http/Controllers/Admin/Setting/SettingController.php:266]: remove `dd`

File: app/Http/Controllers/Auth/RegisterController.php [hunk 3]
```diff
@@ -221,6 +222,7 @@ public function register(Request $request): \Illuminate\Http\JsonResponse|\Illum
         }
         $user = $this->create($request->all());
         event(new Registered($user));
+        Session::put('role_id', 1);
```
Comment (man-git007) [inline app/Http/Controllers/Auth/RegisterController.php:225]: remove static code `1`

File: app/Http/Middleware/OrganizationUserMiddleware.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<?php
+
```
Comment (man-git007) [inline app/Http/Middleware/OrganizationUserMiddleware.php:2]: declare strict type missing

File: app/Http/Middleware/SuperAdminMiddleware.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?php
+
```
Comment (man-git007) [inline app/Http/Middleware/SuperAdminMiddleware.php:2]: declare strict type missing

File: app/IATI/Models/User/User.php [hunk 6]
```diff
@@ -22,6 +22,11 @@ class User extends Authenticatable implements MustVerifyEmail
 {
     use HasApiTokens, HasFactory, Notifiable;
 
+    /**
+     * Role id for superadmin.
+     */
+    public const SUPER_ADMIN_ID = 2;
```
Comment (man-git007) [inline app/IATI/Models/User/User.php:28]: Can this be made dynamic?

File: database/factories/IATI/Models/User/RoleFactory.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Database\Factories\IATI\Models\User;
```
Comment (man-git007) [inline database/factories/IATI/Models/User/RoleFactory.php:3]: declare strict type missing

File: database/migrations/2022_09_13_034205_create_roles_table.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
```
Comment (man-git007) [inline database/migrations/2022_09_13_034205_create_roles_table.php:3]: declare strict type missing

**Review by man-git007 — COMMENTED** [review]
Have added some comments. Please fix them

**Comment by Sanilblank** [comment]
Need to run command:
php artisan create:superadmin

to create a superadmin with the login credentials
Username: superadmin
Password: password

**Pull request merged**